src/js/components/projects/detail.tsx

```diff
@@ -45,112 +43,61 @@ const ProjectDetail = (props: RouteComponentProps) => {
   // This redundant check is used to satisfy TypeScript...
   /* istanbul ignore if */
   if (!product || !project) {
-    return <ProductNotFound />;
+    return <FourOhFour />;
   }
 
   if (
     (productSlug && productSlug !== product.slug) ||
     (projectSlug && projectSlug !== project.slug)
   ) {
-    // Redirect to most recent product slug
+    // Redirect to most recent product/project slug
     return <Redirect to={routes.project_detail(product.slug, project.slug)} />;
```

Comment:
This is project-only, right? But the comment seems to say otherwise. Am I missing something?

Reply:
See the `if`... it redirects if _either_ the product or project slug is outdated. You're correct that it renders project-detail in either case.

Reply:
@winnettjohnston yes this should be project only I think

Reply:
I think it depends how you're parsing the comment. It is redirecting to the most recent product _and_ project slugs, but redirecting to the project-detail URL.